backend/api/ProblemApi/services/movePuils.js

```diff
@@ -0,0 +1,48 @@
+import knex from '~/db/knex';
+import CourseUserIsLearningModel from '~/models/CourseUserIsLearningModel';
+/**
+ * Moves the all "problemUserIsLearning" (Puils) for the given problemIds to the given course Id
+ * @param {*} problemIds the problemIds to move Puils for. 
+ * @param {*} insertedProblems the inserted problems with their new IDs
+ * @param {*} courseId the id of the course to move to
+ */
+const movePuils = async (problemIds, insertedProblems, courseId) => {
+  //get the puils for the given problem ids
+  const puils = await knex('problemUserIsLearning').whereIn('problem_id', problemIds);
+
+  //map the inserted problems to the old problem's puils. 
+  //  We need to do this, because not every problem necessarily has a puil.
+  //  Directly connecting them via problem ID is not possible -> the newly inserted problems have a different id than in the old course.
+  const mappedPuils = puils.map(puil => {
+    return {
+      puil: puil,
+      insertedProblem: insertedProblems[problemIds.indexOf(puil.problemId)]
+    };
+  });
+
+  //move each puil
+  await Promise.all(mappedPuils.map(async (mappedPuil) => await movePuil(mappedPuil, courseId)));
+};
+
+const movePuil = async (mappedPuil, courseId) => {
+  //get the id of the old course the user is learning
+  const oldCuilId = mappedPuil.puil.courseUserIsLearningId;
+  //find the old course the user is learning
+  const oldCuil = (await knex('courseUserIsLearning').where({ id: oldCuilId }))[0];
+  //use the user id and new course id to find the new course the user is learning
+  const newCuil = await CourseUserIsLearningModel.select.oneByCourseIdAndUserId(courseId, oldCuil.userId);
```

Comment:
It's ok to use `CourseUserIsLearningModel` methods, but fyi these are considered legacy, we are switching to `knex` because it's easier to read & more straightforward (direct obvious query is better than `.oneByCourseIdAndUserId()`  that can have anything at all inside).

Reply:
Good to know - makes sense to keep everything on the same level, I'll update accordingly :)